### DISCUSSION TITLE
How to handle possibly undefined objects in initial create

### DISCUSSION THREAD

**Discussion by coltanium13** [pmndrs/zustand#2907] [Q&A] [answered]
Hey @dai-shi , quick question. When creating a store using typescript, how do you handle objects that could be undefined. 

For example, if i have an object in the store, say Person, but it does not have any initial data that i can set it to, and it may never get set at all, do i just do an empty object and cast it? like person: {} as Person, or just leave it as undefined, like person: undefined (or just leave it out of the create object)

Also, if you have a complex object, that does not have any initial data, is it fine to cast the empty object as the type, OR is it best to create a large initial object and take the best logical guesses at the default property values?

thank you!

**Comment by dai-shi** [comment, marked as answer]
> is it fine to cast the empty object as the type

It's not fine if you want to get type checks. I would do this `person: undefined as Person | undefined`.

**Reply by coltanium13** [reply]
OK, that makes sense. So whenever you dont know the initial data for an object or , just make it ``undefined`` inside the create? However, if you know logical defaults for everything, then set them at that time.
